### PR TITLE
Update ChildStderr docs to be clearer

### PR DESCRIPTION
**Pull request by mgattozzi**
Before the docs only had a line about where it was found and that it was
a handle to stderr. This commit changes it so that the summary second line is
removed and that it's a bit clearer about what can be done with it. Part of
#29370

### CODE REVIEW

**Comment by mgattozzi** [comment]
@steveklabnik I'm not sure if this is what you meant by expanding upon the docs in https://github.com/rust-lang/rust/issues/29370. If it is great because I have two PRs ready to go for ChildStdin and ChildStdout, but I figured it best to get a sense for what you wanted this to be expanded too here before opening those two.

File: src/libstd/process.rs [hunk 1]
```diff
@@ -233,8 +233,9 @@ impl fmt::Debug for ChildStdout {
     }
 }
 
-/// A handle to a child process's stderr. This struct is used in the [`stderr`]
-/// field on [`Child`].
+/// A handle to a child process's stderr. It can be used to
```
Comment (GuillaumeGomez) [inline src/libstd/process.rs]: Please keep "This struct is used in the [`stderr`] field on [`Child`]."

**Review by GuillaumeGomez — CHANGES_REQUESTED** [review]

File: src/libstd/process.rs (hunk 1 above)
Comment (steveklabnik) [inline src/libstd/process.rs]: I'm not saying to remove it, I'm saying that there needs to be whitespace between.

**Comment by steveklabnik** [comment]
wow thanks github, dunno how that got duplicated...

File: src/libstd/process.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/libstd/process.rs]: I'd prefer to have it this way as well.

Comment (mgattozzi) [inline src/libstd/process.rs]: Okay will do. That makes more sense.

Comment (mgattozzi) [inline src/libstd/process.rs]: Ahhhh okay got it.

**Comment by mgattozzi** [comment]
I see what your wording in the issue means now. After my other PR is done building locally I can move on to this one.

**Comment by mgattozzi** [comment]
@GuillaumeGomez and @steveklabnik  I've updated the docs to have the newlines and changed it back to the original wording as requested!

**Comment by mgattozzi** [comment]
Awesome! I'll get the other two in soon so that we can knock them off the process list.

**Pull request merged**